internal/pkg/agent/application/monitoring/liveness.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package monitoring
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+)
+
+const formValueKey = "failon"
+
+type LivenessFailConfig struct {
+	Degraded    bool `yaml:"degraded" config:"degraded"`
+	Failed      bool `yaml:"failed" config:"failed"`
+	Coordinator bool `yaml:"coordinator" config:"coordinator"`
+}
+
+// process the form values we get via HTTP
+func handleFormValues(req *http.Request) (LivenessFailConfig, error) {
+	err := req.ParseForm()
+	if err != nil {
+		return LivenessFailConfig{}, fmt.Errorf("Error parsing form: %w", err)
+	}
+
+	defaultUserCfg := LivenessFailConfig{Degraded: false, Failed: false, Coordinator: true}
+
+	for formKey := range req.Form {
+		if formKey != formValueKey {
+			return defaultUserCfg, fmt.Errorf("got invalid HTTP form key: '%s'", formKey)
+		}
+	}
+
+	userConfig := req.Form.Get(formValueKey)
+	switch userConfig {
+	case "failed":
+		return LivenessFailConfig{Degraded: false, Failed: true, Coordinator: true}, nil
+	case "degraded":
+		return LivenessFailConfig{Failed: true, Degraded: true, Coordinator: true}, nil
+	case "heartbeat", "":
+		return defaultUserCfg, nil
+	default:
+		return defaultUserCfg, fmt.Errorf("got unexpected value for `%s` attribute: %s", formValueKey, userConfig)
+	}
+}
+
+func livenessHandler(coord CoordinatorState) func(http.ResponseWriter, *http.Request) error {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
+
+		state := coord.State()
+		isUp := coord.CoordinatorActive(time.Second * 10)
```

Comment:
You could just immediately return StatusServiceUnavailable here, since if this is false none of the form values matter.